File: Lab2/Lab2.java
import java.util.Scanner;

public class Lab2 {
    public static Boolean checkState(String t, String s)
    {
        t = t.replaceAll(" ", "");
        s = s.replaceAll(" ", "");
        int tSize = t.length();
        int sSize = s.length();
        if(tSize == sSize)
        {
            return true;
        }
        else{
            return false;
        }
        
    }

    public static Boolean anagram(String t, String s)
    {
        int size = t.length();
        //if t.lenght == s.length and !t.equalTo(s)
        //  loop through charachters in t and check for the characheters in s
        // if all are true return true;
        Boolean condition = checkState(t, s);
        if(condition)
        {
            for(int i = 0; i < size; i++)
            {
                for(int j = 0; j < size; j++)
                {
                    if(t.charAt(i) == s.charAt(j)){
                        break;
                    }
                    else if (j == size-1 && t.charAt(i) != s.charAt(j))
                    {
                        return false;
                    }
                }
            }
            return true;
        }
        return false;

    }

    public static void main(String[] args)
    {
        String t; 
        String s;
        Scanner scan = new Scanner(System.in);

        System.out.println("first :");
        t = scan.next();
        System.out.println("next: ");
        s = scan.next();

        
        System.out.println(anagram(t, s));
    }






}

File: Lab2/Lab2.py
def check_state(t, s):
    t = t.replace(" ", "")
    s = s.replace(" ", "")
    t_size = len(t)
    s_size = len(s)
    if t_size == s_size:
        return True
    else:
        return False

def anagram(t, s):
    size = len(t)
    condition = check_state(t, s)
    if condition:
        for i in range(size):
            for j in range(size):
                if t[i] == s[j]:
                    break
                elif j == size - 1 and t[i] != s[j]:
                    return False
        return True
    return False

def main():
    t = input("first: ")
    s = input("next: ")
    print(anagram(t, s))

if __name__ == "__main__":
    main()
